JS/cart.js

```diff
@@ -281,25 +143,133 @@ You may also want to check but JSON.parse should automatically parse integer and
     grid_article_in_cart.appendChild(name_cart_div);
     grid_article_in_cart.appendChild(info_cart_div);
     cart_div.appendChild(grid_article_in_cart);
-
     let line = document.createElement('div');
     line.className = "line_in_cart";
     cart_div.appendChild(line);
-    return cart_div;
 }
 /**
  * Function used when we change the quantity of an article to save the changes into the sessionStorage
  * @param {any} article : object that contains the article
  * @param {any} elem : the input that contains the value of the quantity chosen for an article
  */
 function update_json_file(article, elem) {
-    var old_data_saved = JSON.parse(sessionStorage.getItem('articleToCart2'));
-    var tab = [];
-    for (i = 0; i < old_data_saved.length; i++) {
+    'use strict';
+    let old_data_saved = myCart.dataStored;
+    for (let i = 0; i < old_data_saved.length; i++) {
         if (old_data_saved[i].id == article.id) {
             old_data_saved[i].quantity = elem.value;
         }
-        tab.push(old_data_saved[i]);
     }
-    sessionStorage.setItem('articleToCart2', JSON.stringify(tab));
+    sessionStorage.setItem('articleToCart2', JSON.stringify(old_data_saved));
+}
+
+class Cart {
+    constructor(data) {
```

Comment:
I won't add a ton of small comments but only few big general remarks here :
this class is empty and shouldn't the cart should be the one in charge of loading data from the storage, building all the articles (via its class Cart_article) and listen to each and every of them to register when they change to handle the update of what's stored.
The constructor should looks like 
```JS
constructor(place,div){
  this.container = div
  this.data =     sessionStorage.getItem(place);
  let article;  
  for article of data {
    article = new cart_article(article,holder)
    article.addListener(this)
  } 
}
```
The main part of the script should be creating a cart which will then handle his own life allowing for the cart to be embeded in other pages if needed very easly